src/js/components/productSupplier/modals/AddAttributeOptionModal.jsx

```diff
@@ -0,0 +1,96 @@
+import React, { useState } from 'react';
+
+import PropTypes from 'prop-types';
+import { RiCloseFill } from 'react-icons/all';
+import Modal from 'react-modal';
+
+import Button from 'components/form-elements/Button';
+import TextInput from 'components/form-elements/v2/TextInput';
+import useTranslate from 'hooks/useTranslate';
+
+import '../styles.scss';
+
+const AddAttributeOptionModal = ({
+  isOpen,
+  close,
+  selectedAttribute,
+  setValue,
+}) => {
+  const [inputValue, setInputValue] = useState(null);
+  const translate = useTranslate();
+
+  const onChange = (e) => setInputValue(e.target.value);
+
+  const onClose = () => {
+    setInputValue(null);
+    close();
+  };
+
+  const setAttributeValue = (attribute, value) => () => {
+    setValue(`attributes.${selectedAttribute?.id}`, { id: attribute?.id, value, label: value });
+    setInputValue(null);
+    close();
+  };
```

Comment:
I think this doesn't have to be a double callback 
```suggestion
  const setAttributeValue = () => {
    setValue(`attributes.${selectedAttribute?.id}`, { id: selectedAttribute?.id, value, label: inputValue });
    setInputValue(null);
    close();
  };
```
am I wrong?